gaeguli/pipeline.c

```diff
@@ -227,6 +235,11 @@ gaeguli_pipeline_class_init (GaeguliPipelineClass * klass)
       "The internal GStreamer pipeline", GST_TYPE_PIPELINE,
       G_PARAM_READABLE | G_PARAM_STATIC_STRINGS);
 
+  properties[PROP_USE_HW_VAAPI_DECODE] =
+      g_param_spec_boolean ("use-hw-vaapi-decode", "Use HW VAAPI Decoding",
```

Comment:
Nitpick: In "hw-vaapi-decode", "hw" seems redundant to me since VAAPI is _always_ hardware. I suggest shortening the signal name to simple `"use-vaapi-decode"` and also dropping "hw" from any other symbol names you added.

Reply:
As I mentioned in GAEUL PR, https://github.com/hwangsaeul/gaeul/pull/52#discussion_r513852916, the pipeline already has enough information about what jpegdec is needed. For example, when we choose VAAPI, there's no reason to take `nvjpegdec`. 

In my opinion, it is enough to add some logic to choosing the right jpegdec according to GaeguliVideoCodec.

Reply:
Used GaeguliVideoCodec details to set the boolean "use-vaapi-decode". When source agent sets this property, vaapijpegdec is chosen.

Reply:
I'm not so sure making VAAPI decode depend on a target's chosen codec is a good idea, because in the current "1 camera - multiple targets" design this can lead to weird and unobvious behavior:

1. you create a pipeline and add a target with x264 encoder; vsrc will use jpegdec
2. you add another target, this time with VAAPI H.264 encoder; vsrc still uses jpegdec
3. you remove the first target; now you have VAAPI target encoder, but vsrc **still** uses software JPEG decoding

That's not very predictable.

IMO better approach would be
a) have an independent property that controls VAAPI decoding, as it is now
b) automatically choose VAAPI JPEG decoder whenever it's available

As I understand it, your point is "don't add another switch the user could tinker with", in which case the most consistent solution seems b), **always** pick `vaapijpegdec` if the underlying system offers it, revert to software only when hw decoder isn't available.

Reply:
I agreed almost with your point, but there's still an unclear case for me.

We may be able to have various hw decoders regardless they are opens-source or proprietary that we don't know the name of the decoder plugins yet.  In my head,  `nvjpegdec` and `vaapijpegdec` are an example at this time. 
If we take the a) approach, that means we need to declare `use-vaapi-decode` and `use-nv-decode` properties too.
I am not sure how many `use-XXX-decode` property should be added sooner or later.

I think b) is a more user-friendly approach, although it is a little bit vague for programmers. That's the way to delegate responsibility to GStreamer rather than programmers.

Reply:
I agree that b) is a better approach. I shall remove the "use-vaapi-decode" property and elevate the rank of vaapidec if it is available else the the software decoder shall be picked up.

Reply:
"use-vaapi-decode" property is removed. Hardware vaapiipeg decoder is chosen if it is available.